lib/docker_manager/git_repo.rb

```diff
@@ -116,15 +133,19 @@ def upgrade_key
   end
 
   def commit_date(commit)
-    unix_timestamp = run(+'show -s --format="%ct" ' << commit).to_i
+    return nil if commit.blank?
+    unix_timestamp = run("show -s --format='%ct' #{commit}").to_i
     Time.at(unix_timestamp).to_datetime
   end
 
-  def has_origin_main?
-    begin
-      run("branch -a").match?(%r{remotes/origin/main$})
-    rescue StandardError
-      false
+  def origin_branch_exist?(branch_name)
+    return false if branch_name.blank?
+
+    if shallow_clone?
+      branch_name = branch_name.delete_prefix("origin/")
```

Comment:
🏌️ 
```suggestion
      branch_name.delete_prefix!("origin/")
```

Reply:
`branch_name` is a frozen string, so the existing code is probably the most readable solution.